src/inference/ess.jl

```diff
@@ -58,11 +56,6 @@ function AbstractMCMC.step(
     # obtain previous sample
     f = vi[spl]
 
-    # recompute log-likelihood in logp
-    if spl.selector.tag !== :default
-        model(rng, vi, spl)
-    end
```

Comment:
The `spl.selector` field was intended to store the assignment of model parameters to samplers in the Gibbs framework. It was meant to be more efficient than just using the `space` field of Gibbs component samplers since it avoids the model parameters name matching step. It might have other undocumented uses (e.g. determining whether parameters are transformed), so removing this mechanism requires some caution. Is there any reason why we want to remove this mechanism? 

cc @xukai92

Reply:
As far as I remember, currently `spl.selector` is only used to determine if we perform Gibbs sampling or not and rerun the model, i.e., repopulate `vi.logp`, if we do. Since this PR "cleans" the sampling state and moves the Gibbs-specific logic to the Gibbs sampler and `gibbs_rerun`, there is no need for it anymore.

Reply:
So it's not the main purpose of this PR to remove `spl.selector`, rather it disappears automatically when the more fine-tuned (and correct) two-argument logic of `gibbs_rerun` is introduced.